internal/object/test/object.service_test.go

```diff
@@ -28,4 +36,137 @@ func (t *ObjectServiceTest) SetupTest() {
 		BucketName: "mock-bucket",
 		Endpoint:   "mock-endpoint",
 	}
+	t.uploadObjectRequest = &proto.UploadObjectRequest{
+		Filename: "object",
+		Data:     []byte("data"),
+	}
+}
+
+func (t *ObjectServiceTest) TestUploadInternalError() {
+	repo := mock_object.NewMockRepository(t.controller)
+	service := object.NewService(repo, t.conf, t.logger, utils.NewRandomUtils())
+
+	repo.EXPECT().Upload(t.uploadObjectRequest.Data, t.conf.BucketName, gomock.Any()).Return("", "", fmt.Errorf("error"))
+
+	_, err := service.Upload(context.Background(), t.uploadObjectRequest)
+
+	t.EqualError(err, "rpc error: code = Internal desc = Internal server error")
+}
+
+func (t *ObjectServiceTest) TestUpload() {
```

Comment:
the function name should be verbose e.g.
- `TestUploadSuccess`
- `TestUploadInternalError`
- `TestUploadInvalidArgument`